quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -919,6 +921,42 @@ async fn delete_source(
     Ok(())
 }
 
+fn get_source_shards_handler(
+    metastore: MetastoreServiceClient,
+) -> impl Filter<Extract = (impl warp::Reply,), Error = Rejection> + Clone {
+    warp::path!("indexes" / String / "sources" / String / "shards")
+        .and(warp::get())
+        .and(with_arg(metastore))
+        .then(get_source_shards)
+        .and(extract_format_from_qs())
+        .map(into_rest_api_response)
+}
+
+async fn get_source_shards(
+    index_id: IndexId,
+    source_id: SourceId,
+    metastore: MetastoreServiceClient,
+) -> MetastoreResult<Vec<Shard>> {
+    info!(index_id = %index_id, source_id = %source_id, "get-source-shards");
+    let index_metadata_request = IndexMetadataRequest::for_index_id(index_id.to_string());
+    let index_uid: IndexUid = metastore
+        .index_metadata(index_metadata_request)
+        .await?
+        .deserialize_index_metadata()?
+        .index_uid;
+    let shards = metastore
+        .list_shards(ListShardsRequest {
+            subrequests: vec![ListShardsSubrequest {
+                index_uid: Some(index_uid),
+                source_id: source_id.to_string(),
+                ..Default::default()
+            }],
+        })
+        .await?;
+
+    Ok(shards.subresponses[0].clone().shards)
```

Comment:
```suggestion
    Ok(shards.subresponses[0].clone().shards)
```
the clone makes no sense here.